exercise/api/csv/views.py

```diff
@@ -173,6 +174,74 @@ def get_renderer_context(self):
         return context
 
 
+class CourseResultsDataViewSet(NestedViewSetMixin,
+                                 CourseResourceMixin,
+                                 viewsets.ReadOnlyModelViewSet):
+    """
+    List aggregate submission data as data sheet.
+    Following GET parameters may be used to filter submissions:
+    category_id, module_id, exercise_id,
+    filter (N.N where first module id, then optional exercise ids)
+    """
+    # submission_count, total_points, max_points, (time_usage) / exercise / chapter / module
+    permission_classes = api_settings.DEFAULT_PERMISSION_CLASSES + [
+        IsCourseAdminOrUserObjIsSelf,
+    ]
+    filter_backends = (
+        IsCourseAdminOrUserObjIsSelf,
+    )
+    renderer_classes = [
+        CSVRenderer,
+    ] + api_settings.DEFAULT_RENDERER_CLASSES
+    lookup_field = 'user_id'
+    lookup_url_kwarg = 'user_id'
+    lookup_value_regex = REGEX_INT_ME
+    parent_lookup_map = {'course_id': 'enrolled.id'}
+    queryset = UserProfile.objects.all()
+
+    def get_search_args(self, request):
+        return {
+            'number': request.GET.get('filter'),
+            'category_id': int_or_none(request.GET.get('category_id')),
+            'module_id': int_or_none(request.GET.get('module_id')),
+            'exercise_id': int_or_none(request.GET.get('exercise_id')),
+            'filter_for_assistant': not self.is_teacher,
+       }
+
+    def list(self, request, version=None, course_id=None):
+        profiles = self.filter_queryset(self.get_queryset())
+        return self.serialize_profiles(request, profiles)
+
+    def retrieve(self, request, version=None, course_id=None, user_id=None):
+        return self.serialize_profiles(request, [self.get_object()])
+
+    def serialize_profiles(self, request, profiles):
+        search_args = self.get_search_args(request)
+        entry, exercises = self.content.search_entries(**search_args)
+        ids = [e['id'] for e in exercises if e['type'] == 'exercise']
+        exclude_list = [Submission.STATUS.ERROR, Submission.STATUS.REJECTED]
+        if(request.GET.get('show_unofficial') != 'true'):
+            exclude_list.append(Submission.STATUS.UNOFFICIAL)
+        aggr = Submission.objects\
```

Comment:
Good point, but in this case I don't think it makes a difference. There is the row .values('submitters__user_id','exercise_id') which results in the mentioned fields to be excluded anyway. Below is the generated SQL query:

```
SELECT "userprofile_userprofile"."user_id",
       "exercise_submission"."exercise_id",
       MAX("exercise_submission"."grade") AS "total",
       COUNT("exercise_submission"."id") AS "count"
FROM "exercise_submission"
INNER JOIN "exercise_submission_submitters" ON ("exercise_submission"."id" = "exercise_submission_submitters"."submission_id")
INNER JOIN "userprofile_userprofile" ON ("exercise_submission_submitters"."userprofile_id" = "userprofile_userprofile"."id")
WHERE ("exercise_submission"."exercise_id" IN (
...
)
       AND "exercise_submission_submitters"."userprofile_id" IN
         (SELECT U0."id"
          FROM "userprofile_userprofile" U0
          INNER JOIN "course_enrollment" U1 ON (U0."id" = U1."user_profile_id")
          INNER JOIN "course_enrollment" U3 ON (U0."id" = U3."user_profile_id")
          WHERE (U1."course_instance_id" = 1
                 AND U3."course_instance_id" = 1))
       AND NOT ("exercise_submission"."status" IN (error,
                                                   rejected,
                                                   unofficial)))
GROUP BY "userprofile_userprofile"."user_id",
         "exercise_submission"."exercise_id"
```

Reply:
I checked, just in case, that adding .defer("feedback","assistant_feedback","submission_data","grading_data")\ after the .values() resulted in exactly the same query.